packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -544,40 +496,84 @@ export function TransferPanel() {
     }
   }
 
-  const transfer = async () => {
-    const signerUndefinedError = 'Signer is undefined'
-
-    try {
-      setTransferring(true)
-      if (chainId !== networks.sourceChain.id) {
-        await switchNetworkAsync?.(networks.sourceChain.id)
-      }
-    } finally {
-      setTransferring(false)
+  async function hasDestinationAddressError() {
+    const destinationAddressError = await getDestinationAddressError({
+      destinationAddress,
+      isSmartContractWallet,
+      isTeleportMode
+    })
+    if (destinationAddressError) {
+      console.error(destinationAddressError)
```

Comment:
I don't think this is needed, function like this shouldn't produce side effects like this imho

Reply:
We can just
```typescript
return typeof destinationAddressError !== null
```

Reply:
But also not sure if we need a helper, we could just use `useState` for this, and then we can use `useMemo` for `isTransferAllowed`

Reply:
i refactored the code for destination address error across files